include/zephyr/arch/arch_interface.h

```diff
@@ -46,6 +46,8 @@ typedef struct z_thread_stack_element k_thread_stack_t;
 
 typedef void (*k_thread_entry_t)(void *p1, void *p2, void *p3);
 
+typedef struct __esf z_arch_esf_t;
```

Comment:
Style: since this is always used in the API by value (and is an arch-specific opaque type anyway) our coding guidelines would demand that it be passed as a struct type.  Since you're cleaning up the API otherwise, might be a good place to fix up.  Maybe call this `struct arch_esf` or `struct arch_exception` or something similar?

I guess we'd need some deprecation strategy for the old typedef; despite being a "z_" internal thing it's always been part of the k_sys_fatal_error_handler() argument list and exposed to apps.

Reply:
I'd go for `struct arch_esf` so that it is more recognizable with the existing `z_arch_esf_t`, or probably just adopt Linux's `struct pt_regs`